### ISSUE TITLE
Redundant "ToString()" call

### ISSUE THREAD

**Issue by tvsonar** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#712]
R# _RedundantToStringCall_

The compiler automatically calls `ToString` if we are concatenating values, or if we are passing the expression to a `string.Format`.

```C#
var s = "" + 1.ToString(); //Noncompliant
s = string.Format("{0}", 1.ToString()); //Noncompliant
```

**Comment by ganncamp** [comment]
https://jira.sonarsource.com/browse/RSPEC-1858

**Comment by tvsonar** [comment]
TODO: check for other cases

**Comment by ganncamp** [comment]
https://jira.sonarsource.com/browse/RSPEC-1858 ?

**Comment by tvsonar** [comment]
With a title change and a subtask that should be okay.

**Comment by ganncamp** [comment]
"toString()" should not be called on a String or when the compiler would do it automatically
